src/v/utils/extended_rwlock.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "seastarx.h"
+#include "ssx/semaphore.h"
+#include <seastar/core/rwlock.hh>
+
+class extended_rwlock;
+
+class rwlock_unit {
+    extended_rwlock* _lock;
+    bool _is_write_lock{false};
+    bool _is_moved{false};
+
+public:
+    rwlock_unit(
+      extended_rwlock* lock,
+      bool is_write_lock) noexcept
+      : _lock(lock)
+      , _is_write_lock(is_write_lock) {}
+    rwlock_unit(rwlock_unit&& token) noexcept
+      : _lock(token._lock)
+      , _is_write_lock(token._is_write_lock) {
+        token._is_moved = true;
+    }
+    rwlock_unit(const rwlock_unit&) = delete;
+    rwlock_unit() = delete;
+
+    ~rwlock_unit() noexcept;
+};
+
+class extended_rwlock {
+public:
+    extended_rwlock(ss::lw_shared_ptr<ss::basic_rwlock<>> lock)
```

Comment:
Can't we just extend ss::basic_rwlock(), passing a lock with shared_ptr seems a little off. probably just call it ssx::rwlock or something (extended seems unnecessary)

Reply:
done